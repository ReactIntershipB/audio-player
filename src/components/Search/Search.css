.searcher-wrapper {
    margin-top: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.searcher-wrapper .ant-input-group-wrapper {
    width: 30%;
}

.dropdown {
    margin: 0 20px;
}

.ant-dropdown .ant-dropdown-menu {
    padding: 0;
    margin-top: 5px;
}

.ant-dropdown-link,
.ant-dropdown .ant-dropdown-menu .ant-dropdown-menu-item {
    width: 90px;
}