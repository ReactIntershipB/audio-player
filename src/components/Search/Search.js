import React, { Component } from 'react';
import { Input, Dropdown, Icon, Menu, Button } from 'antd';
import { observer, inject } from 'mobx-react';
import PropTypes from 'prop-types';

import './Search.css';

@inject('searchModel', 'appUI')
@observer
class Search extends Component {
    componentDidMount() {
      this.fetchData();
    }

    componentDidUpdate(prevProps) {
      // if (this.props.searchModel.term !== prevProps.term || this.props.searchModel.filterName !== prevProps.filterName) {
      //   this.fetchData(this.props.searchModel.term, this.props.searchModel.filterName);
      // }
    }

    get dropdownMenu() {
        return (
            <Menu
                className='dropdown-menu'
                value={this.props.searchModel.filterName}
                onClick={this.onFilterClick}
                selectedKeys={[this.props.searchModel.filterName]}
            >
            <Menu.Item key="artist" value='artist'>
                Artist
            </Menu.Item>
            <Menu.Item key="album" value='album'>
                Album
            </Menu.Item>
            <Menu.Item key="track" value='track'>
                Track
            </Menu.Item>
        </Menu>
        );
    };

    get addonSearchIcon() {
        return (
            <Icon type="search" />
        );
    };

    fetchData = (term, filterName) => {
      if (term && filterName) {
        this.props.searchModel.find(this.props.match(term, filterName));
      }
    }

    getFilterName = (filterName) => {
      return filterName.charAt(0).toUpperCase() + filterName.slice(1);
    }

    onInputChange = e => {
      this.props.searchModel.inputChange(e.target.value);
      this.props.appUI.enableButton(e.target.value);
    }

    onFilterClick = e => {
      this.props.searchModel.filterChange(e.item.props.value);
    }

    onSubmitClick = e => {
      e.preventDefault();
      const { term, filterName } = this.props.searchModel;
      if (term !== '') {
        this.props.searchModel.find(term, filterName);
        this.props.searchModel.setTermText(term);
        this.clearInput();
        this.props.history.push(`/search/${filterName}`);
      }
    }

    clearInput = () => {
      this.props.searchModel.term = '';
    }

    render() {
      return (
        <form className='searcher-wrapper' onSubmit={this.onSubmitClick}>
          <Input
            placeholder="Find some music..."
            className="searcher-input"
            onChange={this.onInputChange}
            value={this.props.searchModel.term}
            addonAfter={this.addonSearchIcon}
          />
          <Dropdown
            className='dropdown'
            overlay={this.dropdownMenu}
          >
            <Button className="ant-dropdown-link">
              {this.getFilterName(this.props.searchModel.filterName)}
              <Icon type="down" />
            </Button>
          </Dropdown>
          <Button
            type="primary"
            disabled={this.props.appUI.isButtonDisabled}
            onClick={this.onSubmitClick}
          >
            Submit
          </Button>
        </form>
      );
    }
}

Search.propTypes = {
  searchModel: PropTypes.object,
  appUI: PropTypes.object,
  history: PropTypes.object,
  match: PropTypes.object
};

export default Search;
