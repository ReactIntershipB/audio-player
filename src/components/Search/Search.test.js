import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';

import Search from './Search';

class MockSearchModel {
    term = 'test';
    termText = 'test';
    filterName = 'artist';
}

describe('Search', () => {
    it('should match snapshot', () => {
        // arrange
        const props = {
            albumModel: {},
            songModel: {},
            searchModel: new MockSearchModel(),
            appUI: {}
        };
        const match = {
            params: {}
        };

        // act
        const search = renderer
            .create(
                <Provider {...props} >
                    <Search match={match} />
                </Provider>
            ).toJSON();

        // assert
        expect(search).toMatchSnapshot();
    });
});
