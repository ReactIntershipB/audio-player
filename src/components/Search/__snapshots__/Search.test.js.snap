// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div
  className="search-container"
>
  <div
    className="search-form-container"
  >
    <span
      className="ant-input-search search-input ant-input-search-enter-button ant-input-affix-wrapper"
      style={undefined}
    >
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Find your favourite artist"
        size={undefined}
        style={null}
        type="text"
        value="test"
      />
      <span
        className="ant-input-suffix"
      >
        <button
          className="ant-btn ant-input-search-button ant-btn-primary"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          <i
            className="anticon anticon-search"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="search"
              fill="currentColor"
              height="1em"
              onClick={undefined}
              style={undefined}
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M608 112c-167.9 0-304 136.1-304 304 0 70.3 23.9 135 63.9 186.5L114.3 856.1a8.03 8.03 0 0 0 0 11.3l42.3 42.3c3.1 3.1 8.2 3.1 11.3 0l253.6-253.6C473 696.1 537.7 720 608 720c167.9 0 304-136.1 304-304S775.9 112 608 112zm161.2 465.2C726.2 620.3 668.9 644 608 644s-118.2-23.7-161.2-66.8C403.7 534.2 380 476.9 380 416s23.7-118.2 66.8-161.2c43-43.1 100.3-66.8 161.2-66.8s118.2 23.7 161.2 66.8c43.1 43 66.8 100.3 66.8 161.2s-23.7 118.2-66.8 161.2z"
              />
            </svg>
          </i>
        </button>
      </span>
    </span>
    <div
      className="ant-radio-group ant-radio-group-solid radio-group-input"
      id={undefined}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      style={undefined}
    >
      <label
        className="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button ant-radio-button-checked"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={true}
            className="ant-radio-button-input"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="artist"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Artist
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={false}
            className="ant-radio-button-input"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="album"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Album
        </span>
      </label>
      <label
        className="ant-radio-button-wrapper"
        onMouseEnter={undefined}
        onMouseLeave={undefined}
        style={undefined}
      >
        <span
          className="ant-radio-button"
          style={Object {}}
        >
          <input
            autoFocus={undefined}
            checked={false}
            className="ant-radio-button-input"
            disabled={false}
            id={undefined}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={undefined}
            onFocus={[Function]}
            readOnly={undefined}
            tabIndex={undefined}
            type="radio"
            value="track"
          />
          <span
            className="ant-radio-button-inner"
          />
        </span>
        <span>
          Track
        </span>
      </label>
    </div>
  </div>
</div>
`;
